components/Chips/src/MDCChipView.m

```diff
@@ -450,49 +450,29 @@ - (void)setContentHorizontalAlignment:(UIControlContentHorizontalAlignment)align
 }
 
 - (void)updateTitleFont {
-  UIFont *customTitleFont = _titleFont;
-
   // If we have a custom font apply it to the label.
   // If not, fall back to the Material specified font.
-  if (customTitleFont) {
-    // If we are automatically adjusting for Dynamic Type resize the font based on the text style
-    if (_mdc_adjustsFontForContentSizeCategory) {
-      self.titleLabel.font = [customTitleFont
-          mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
-                       scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
-    } else {
-      self.titleLabel.font = customTitleFont;
-    }
-  } else {
-    // TODO(#2709): Migrate to a single source of truth for fonts
-    // There is no custom font, so use the default font.
-    if (_mdc_adjustsFontForContentSizeCategory) {
-      // If we are using the default (system) font loader, retrieve the
-      // font from the UIFont preferredFont API.
-      if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
-        _titleLabel.font = [UIFont mdc_preferredFontForMaterialTextStyle:kTitleTextStyle];
-      } else {
-        // There is a custom font loader, retrieve the font and scale it.
-        UIFont *customTypographyFont = [MDCTypography buttonFont];
-        _titleLabel.font = [customTypographyFont
-            mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
-                         scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
-      }
-    } else {
-      // If we are using the default (system) font loader, retrieve the
-      // font from the UIFont standardFont API.
-      if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
-        _titleLabel.font = [UIFont mdc_standardFontForMaterialTextStyle:kTitleTextStyle];
-      } else {
-        // There is a custom font loader, retrieve the font from it.
-        _titleLabel.font = [MDCTypography buttonFont];
-      }
-    }
+  UIFont *titleFont = _titleFont ?: [[self class] defaultTitleFont];
+
+  // If we are automatically adjusting for Dynamic Type resize the font based on the text style
+  if (self.mdc_adjustsFontForContentSizeCategory) {
+    titleFont =
+        [titleFont mdc_fontSizedForMaterialTextStyle:kTitleTextStyle
+                                scaledForDynamicType:_mdc_adjustsFontForContentSizeCategory];
   }
+  self.titleLabel.font = titleFont;
 
   [self setNeedsLayout];
 }
 
++ (UIFont *)defaultTitleFont {
+  // TODO(#2709): Migrate to a single source of truth for fonts
+  if ([MDCTypography.fontLoader isKindOfClass:[MDCSystemFontLoader class]]) {
+    return [UIFont mdc_standardFontForMaterialTextStyle:kTitleTextStyle];
```

Comment:
```
if (self.mdc_adjustsFontForContentSizeCategory) {
  return [UIFont mdc_preferredFontForMaterialTextStyle:kTitleTextStyle];
} else {
  return [UIFont mdc_standardFontForMaterialTextStyle:kTitleTextStyle];
}
```

We use to call mdc_preferredFontForMaterialTextStyle:kTitleTextStyle when the user set _mdc_adjustsFontForContentSizeCategory

Reply:
I feel mdc_adjustsFontForContentSizeCategory doesn't need to be called in the defaultTitleFont, because it is default? If we add it, there is no way to get a non scaled font.

Reply:
wenyu walked me through this. 
mdc_fontSizedForMaterialTextStyle:kTitleTextStyle  actually checks that condition in it's branching flows.